pkg/kubectl/cmd/cmd.go

```diff
@@ -145,6 +146,21 @@ func runHelp(cmd *cobra.Command, args []string) {
 	cmd.Help()
 }
 
+func calledByAlias(cmd *cobra.Command) bool {
+	if len(cmd.Aliases) == 0 {
+		return false
+	}
+	for _, arg := range os.Args {
+		if arg == cmd.Name() {
+			return false
+		}
+		if cmd.HasAlias(arg) {
```

Comment:
You are right, it will fail if a subcommand has a conflicting name. It will fail in other cases like `kubectl -f fileWithConfilictingName cmd.Name`. We essentially need to repeat what Find() does if we want this check works 100% correctly. 

No, cobra does not give any clue on whether the command is called by alias:
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/Godeps/_workspace/src/github.com/spf13/cobra/command.go#L399

My implementation has false positive/false negative errors in rare cases, but it will not cause panic. And because the deprecated warning things are going to be short lived, I think we can accept this hacky implementation for the time being, as it at least fixes `go test` errors.